### ISSUE TITLE
How does CatBoost encode categorical variables on test set?

### ISSUE THREAD

**Issue by minegazioglu** [catboost/catboost#2001]
I know how catboost encodes categorical variables on training set? But I have not seen a clear explanation on how it encodes the categorical variables on the test set. I have seen some github issues regarding this issue but the answers were not sufficient. I have a theory that on the test set catboost just does regular target encoding on the training set and replaces these target statistics on the test set. If I am correct, how are the new levels handled in the test set? If I am not correct, can someone explain how the categorical variables are handled in the test set in detail? The papers or the catboost documentation also does not reveal much on the issue.
